langchain4j-core/src/main/java/dev/langchain4j/data/document/Metadata.java

```diff
@@ -65,26 +61,45 @@ public Metadata() {
      * Constructs a Metadata object from a map of key-value pairs.
      *
      * @param metadata the map of key-value pairs; must not be {@code null}. {@code null} values are not permitted.
-     *                 Supported value types: {@link String}, {@link Integer}, {@link Long}, {@link Float}, {@link Double}
+     *                 Supported value types: {@link String}, {@link Integer}, {@link Long}, {@link Float}, {@link Double}, {@link Collection}
```

Comment:
```suggestion
     *                 Supported value types: {@link String}, {@link Integer}, {@link Long}, {@link Float}, {@link Double}, and {@link Collection} of the former types.
```

Reply:
If I understand correctly there are meaningful use cases only for collections of `String`s and `UUID`s? If that's correct should we enforce this also in the collection validation?

Reply:
@mariofusco My access control use case (#1905) is fully covered by String and UUID types.
@Boonichi what about storing points in Qdrant?

Currently only getter for Collection<String> and Collection<UUID> exist, but it would be easy to implement them for all supported types.

Reply:
For my cases, support for the String and UUID types is enough